test/functional/settings-e2e-specs.js

```diff
@@ -0,0 +1,49 @@
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import _ from 'lodash';
+import AndroidDriver from '../..';
+import DEFAULT_CAPS from './desired';
+import { sleep } from 'asyncbox';
+
+chai.should();
+chai.use(chaiAsPromised);
+
+let defaultCaps = _.defaults({
+  androidInstallTimeout: 90000,
+  browserName: 'chrome'
+}, DEFAULT_CAPS);
+
+describe('toggle wifi tests', () => {
+  let driver;
+
+  describe('functional', () => {
+    before(() => {
```

Comment:
`this` is only the mocha context if it is in a normal function. So this needs to be 
```js
before(function () {
```